pkg/kncloudevents/message_sender_test.go

```diff
@@ -180,6 +184,86 @@ func TestHTTPMessageSenderSendWithRetries(t *testing.T) {
 	}
 }
 
+func TestRetriesOnNetworkErrors(t *testing.T) {
+
+	port := rand.Int31n(math.MaxUint16-1024) + 1024
+	n := int32(10)
```

Comment:
I don't think you need all the `int32`.

Reply:
I use `n` to set `DeliverySpec.Retry` which is an `int32` and then I compare it with `nCalls`, so instead of a cast when I compare them I just use the same type for both, does that make sense?

Reply:
Ah, makes sense. I thought you are just doing that in n and nCalls. I personally would do the cast once in `Int32Ptr`, but it is just me.

Reply:
What's the approximate run time for this test? I think recently the linear backoff is changed to: `t, 2t, 3t, 4t`, instead of `t, t, t, t` for the interval.

Reply:
I attached test logs in the PR body, it ran in `3.61s`.
I use the smallest backoff supported by the library but we can reduce `n` if that's too much for a single unit test.